manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/services/ProcessorServiceImpl.java

```diff
@@ -288,15 +295,20 @@ public Processor updateProcessorStatus(ProcessorStatusDTO statusDTO) {
                 if (isOperationComplete(updatedConditions)) {
                     if (Objects.isNull(processor.getPublishedAt())) {
                         processor.setPublishedAt(ZonedDateTime.now(ZoneOffset.UTC));
-                        // TODO: record metrics with MetricsService
-                        // metricsService.onOperationComplete(processor, MetricsOperation.MANAGER_RESOURCE_PROVISION);
+                        processor.setPublishedAt(ZonedDateTime.now(ZoneOffset.UTC));
+                        metricsService.onOperationComplete(processor, MetricsOperation.MANAGER_RESOURCE_PROVISION);
                     }
+                } else if (isOperationFailed(updatedConditions)) {
+                    metricsService.onOperationFailed(processor, MetricsOperation.MANAGER_RESOURCE_PROVISION);
                 }
                 break;
 
             case UPDATE:
-                // TODO: record metrics with MetricsService
-                // metricsService.onOperationComplete(processor, MetricsOperation.MANAGER_RESOURCE_MODIFY);
+                if (isOperationComplete(updatedConditions)) {
+                    metricsService.onOperationComplete(processor, MetricsOperation.MANAGER_RESOURCE_MODIFY);
+                } else if (isOperationFailed(updatedConditions)) {
+                    metricsService.onOperationFailed(processor, MetricsOperation.MANAGER_RESOURCE_MODIFY);
+                }
```

Comment:
According to the current update strategy we are going to receive the updates every X seconds, so you would register the success/failure operations multiple times - the metrics would not have much sense anymore

Reply:
I hear you!! 

`CREATE` and `DELETE` will only be recorded once due to other checks; but `UPDATE` and `FAILED` are problematic.

I wonder what can be done to address `UPDATE` and `FAILED` (I don't really want to compare `Condition`'s).

Hmmm.....

Reply:
@r00ta I've added some more commits.

Most notably is the last that prevents duplicate updates from being recorded.

I convert `Condition`'s to `ManagedResourceStatus` when checking for duplicate updates to prevent failures in different dependencies from being recorded as successive failed operations. I think it is an improvement on what happens in `v1` (where we potentially record duplicate metrics for failures).